File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/pumpkin/EquipableCursedCarvedPumpkinBlock.java
package generations.gg.generations.core.generationscore.common.world.level.block.pumpkin;

import net.minecraft.world.entity.EquipmentSlot;
import net.minecraft.world.item.Equipable;
import org.jetbrains.annotations.NotNull;

public class EquipableCursedCarvedPumpkinBlock extends CursedCarvedPumpkinBlock implements Equipable {
    public EquipableCursedCarvedPumpkinBlock(Properties arg) {
        super(arg);
    }

    @Override
    public @NotNull EquipmentSlot getEquipmentSlot() {
        return EquipmentSlot.HEAD;
    }
}

File: src/common/main/kotlin/generations/gg/generations/core/generationscore/common/world/level/block/pumpkin/EquipableCursedCarvedPumpkinBlock.kt
package generations.gg.generations.core.generationscore.common.world.level.block.pumpkin

import net.minecraft.world.entity.EquipmentSlot
import net.minecraft.world.item.Equipable

class EquipableCursedCarvedPumpkinBlock(arg: Properties) : CursedCarvedPumpkinBlock(arg), Equipable {
    override fun getEquipmentSlot(): EquipmentSlot = EquipmentSlot.HEAD
}
